tests/Builder/Expression/DateAddOperatorTest.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder\Expression;
+
+use MongoDB\Builder\Expression;
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Query;
+use MongoDB\Builder\Stage;
+use MongoDB\Tests\Builder\PipelineTestCase;
+
+/**
+ * Test $dateAdd expression
+ */
+class DateAddOperatorTest extends PipelineTestCase
+{
+    public function testAddAFutureDate(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::project(
+                expectedDeliveryDate: Expression::dateAdd(
+                    startDate: Expression::dateFieldPath('purchaseDate'),
+                    unit: 'day',
```

Comment:
It would be very useful to have a enum for this units. https://jira.mongodb.org/browse/PHPLIB-1370